Update Blazor Templates #13200 (#13536)
aspnet/AspNetCore.Docs: aspnetcore/blazor/hosting-models.md

![Blazor client-side: The Blazor app runs on a UI thread inside the browser.](hosting-models/_static/client-side.png)

> To create a Blazor app using the client-side hosting model, use the following template:

Reviewer: Let's try to avoid the dreaded *single bullet list*. :smile:

```suggestion
To create a Blazor app using the client-side hosting model, use the **Blazor WebAssembly App** template ([dotnet new blazorwasm](/dotnet/core/tools/dotnet-new)).
```
```

Reviewer: Also, we need to get the `--hosted` (is that correct?) switch in here. You might want to keep the list but recast it a bit so that the 1st bullet is for pure client-side and the 2nd is for client-side hosted.

> * **Blazor WebAssembly App** ([dotnet new blazorwasm](/dotnet/core/tools/dotnet-new)) &ndash; Deployed as a set of static files.

Reviewer: Remove and tighten spacing between last paragraph and next.

> In the above template, there is a **Hosted** option which configiures the application to use a ASP.NET Core server backend. The ASP.NET Core app serves the Blazor app to clients. The Blazor client-side app can interact with the server over the network using web API calls or [SignalR](xref:signalr/introduction).

Reviewer: We can't say "above" ... they don't permit *directional* terms. You can say it like this tho ...

```suggestion
After selecting the **Blazor WebAssembly App** template, select the **Hosted** option to configure the app to use an ASP.NET Core server backend. The ASP.NET Core app serves the Blazor app to clients. The Blazor client-side app can interact with the server over the network using web API calls or [SignalR](xref:signalr/introduction).
```

![Blazor client-side: The Blazor app runs on a UI thread inside the browser.](hosting-models/_static/client-side.png)

> To create a Blazor app using the client-side hosting model, use the **Blazor WebAssembly App** template ([dotnet new blazorwasm](/dotnet/core/tools/dotnet-new)). After selecting the **Blazor WebAssembly App** template, select the **Hosted** option to configure the app to use an ASP.NET Core server backend ([dotnet new blazorwasm --hosted](/dotnet/core/tools/dotnet-new)). The ASP.NET Core app serves the Blazor app to clients. The Blazor client-side app can interact with the server over the network using web API calls or [SignalR](xref:signalr/introduction).

Reviewer: This is for discussion ... I liked it as two paragraphs to make it clearer that the hosted option is just that ... an option. I add a hair more text to make that point clear. This is just for discussion thus far ... I'm not fully married to the exact language. I'm just worried that peeps will get confused about it.

```suggestion
To create a Blazor app using the client-side hosting model, use the **Blazor WebAssembly App** template ([dotnet new blazorwasm](/dotnet/core/tools/dotnet-new)).

After selecting the **Blazor WebAssembly App** template, you have the option of configuring the app to use an ASP.NET Core backend by selecting the **Hosted** option ([dotnet new blazorwasm --hosted](/dotnet/core/tools/dotnet-new)). The ASP.NET Core app serves the Blazor app to clients. The Blazor client-side app can interact with the server over the network using web API calls or [SignalR](xref:signalr/introduction).
```

Author: I think we can call out the fact that there are 3 versions of Blazor

- Server-Side Blazor (the artist formerly known as Razor Components)
- Web Assembly Blazor
  - Standalone single project template
  - ASP.NET Core Hosted 3 project template

But this is represented by 2 actual templates, one having some additional configuration which is optional

Reviewer: Sounds good ... my only concern was that if we run the *hosted* option right on the back of the *non-hosted* option (same paragraph and with the instruction to "After selecting the **Blazor WebAssembly App** template, select ...") that it gets a little confusing and almost sounds like the *hosted* option must be used.

Author: sure, defer to @danroth27 on intended language

Reviewer 2: @guardrex 's suggestion looks good to me.

Reviewer 2: I think though that the `--hosted` option shows up as "ASP.NET Core hosted" in VS. Will folks using VS understand what we mean by the "Hosted" option?

Reviewer: It's a check box, right? If it is, I can change my suggestion to say ...

> ... backend by selecting the **Hosted** check box when creating the app ([dotnet new blazorwasm --hosted](/dotnet/core/tools/dotnet-new)).

Reviewer: ```suggestion
To create a Blazor app using the client-side hosting model, use the **Blazor WebAssembly App** template ([dotnet new blazorwasm](/dotnet/core/tools/dotnet-new)).

After selecting the **Blazor WebAssembly App** template, you have the option of configuring the app to use an ASP.NET Core backend by selecting the **Hosted** check box ([dotnet new blazorwasm --hosted](/dotnet/core/tools/dotnet-new)). The ASP.NET Core app serves the Blazor app to clients. The Blazor client-side app can interact with the server over the network using web API calls or [SignalR](xref:signalr/introduction).
```

Reviewer 2: @guardrex Yeah, it's a checkbox, but the label for the checkbox is "ASP.NET Core hosted"

Reviewer: Ah ... gotcha. I'll update that suggestion :point_up:.

I really, **_really_** need to get Aquent to send me a laptop. 🤦‍♂

I'll do that ... I'll send them a message now.

Note to self on next desktop build ... **_MASSIVE STORAGE! Plz!_**. lol
